### ISSUE TITLE
[BUG] High CPU usage by Instance Managers

### ISSUE THREAD

**Issue by moray95** [longhorn/longhorn#3874]
## Describe the bug

Som of Longhorn's Instance Managers use very high CPU, causing resources to exhaust, volumes not being able to create/attach/detach, causing snapshot deletes to take very long times, the Longhorn UI getting stuck, mostly because of timeout issues. The CPU usage can sometimes go above 80% of a node's capacity, which also causes other workloads to struggle.

## To Reproduce

I'm not sure of the exact steps for reproducing the issue, but it became worse after performing a graceful drain + reboot on 3 of our nodes for maintenance.

## Expected behavior

Instance Manager should not use that much CPU.

## Environment

 - Longhorn version: 1.2.4
 - Installation method (e.g. Rancher Catalog App/Helm/Kubectl): Rancher Catalog
 - Kubernetes distro (e.g. RKE/K3s/EKS/OpenShift) and version: RKE 1.21.6
   - Number of management node in the cluster: 3
   - Number of worker node in the cluster: 8
 - Node config
   - OS type and version: Ubuntu 20.04.4 LTS
   - CPU per node: 12
   - Memory per node: 64GB
   - Disk type(e.g. SSD/NVMe): SSD
   - Network bandwidth between the nodes: 1 Gb/s
 - Underlying Infrastructure (e.g. on AWS/GCE, EKS/GKE, VMWare/KVM, Baremetal): VMWare
 - Number of Longhorn volumes in the cluster: 22

## Additional context

Last 24h Instance Manager CPU usage
<img width="2922" alt="Screen Shot 2022-04-20 at 11 46 43" src="https://user-images.githubusercontent.com/3223697/164191539-3b9e711d-3c8f-4d3b-bbe1-7185e371dacf.png">

**Comment by derekbit** [comment]
Are there rebuild or backup tasks while hitting the high CPU usage?

**Comment by moray95** [comment]
Not necessarily, we've seen high usage both during rebuilt and not. We dropped most of or volume's replicas from 2 to 1 to minimize rebuilds. We used to have recurring a job to backup PVC's, but we temporarily stopped it. None showed any noticeable drop on CPU usage.

**Comment by gaow1423** [comment]
the similar issue here https://github.com/longhorn/longhorn/issues/3889#issue-1213795826

**Comment by mike12806** [comment]
Possibly same issue? https://github.com/longhorn/longhorn/issues/6695
